packages/ui/src/components/va-tree-view/components/VaTreeNode/VaTreeNode.vue

```diff
@@ -0,0 +1,229 @@
+<template>
+  <div
+    class="va-tree-node"
+    role="treeitem"
+    :aria-expanded="$props.node.expanded"
+    :aria-disabled="$props.node.disabled"
+    :aria-checked="$props.node.checked"
+    :tabindex="tabIndexComputed"
+    @keydown.right.stop.prevent="handleToggleNode($event, $props.node)"
+    @keydown.left.stop.prevent="handleToggleNode($event, $props.node)"
+  >
+    <div
+      class="va-tree-node-root"
+      @click.stop="toggleNode($props.node)"
+    >
+      <div class="va-tree-node-content" :class="gapClassComputed">
+        <div
+          v-if="$props.node.hasChildren"
+          class="va-tree-node-content__item va-tree-node-content__item--leaf"
+        >
+          <slot v-bind="$props.node" name="icon-toggle">
+            <va-icon
+              :name="isExpandedComputed ? 'keyboard_arrow_down' : 'keyboard_arrow_right'"
+              size="20px"
+            />
+          </slot>
+        </div>
+        <div
+          v-if="selectable"
+          class="va-tree-node-content__item"
+          @click.stop
+        >
+          <slot name="checkbox">
+            <va-checkbox
+              v-model="$props.node.checked"
+              :color="colorComputed"
+              @update:model-value="(isChecked) => toggleCheckbox($props.node, isChecked)"
+              class="va-tree-node__checkbox"
+            />
+          </slot>
+        </div>
+        <div v-if="hasIconComputed" class="va-tree-node-content__item">
+          <slot name="icon" v-bind="$props.node"></slot>
+        </div>
+        <div class="va-tree-node-content__body">
+          <slot name="content" v-bind="$props.node">{{ labelComputed }}</slot>
+        </div>
+      </div>
+    </div>
+    <div
+      v-show="$props.node.hasChildren"
+      role="group"
+      :aria-hidden="!$props.node.expanded"
+      class="va-tree-node-children"
+      :class="expandedClassComputed"
+    >
+      <va-tree-node
+        v-for="childNode in $props.node.children"
+        :key="getKey(childNode)"
+        :node="childNode"
+      >
+        <template v-for="(_, name) in $slots" :key="name" v-slot:[name]="slotScope">
+          <slot :name="name" v-bind="slotScope" />
+        </template>
+      </va-tree-node>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, inject, computed, PropType } from 'vue'
+
+import { useBem } from '../../../../composables'
+
+import { VaIcon, VaCheckbox } from '../../../'
+import { TreeViewKey, TreeNode } from '../../types'
+
+export default defineComponent({
+  name: 'VaTreeNode',
+
+  props: {
+    node: {
+      type: Object as PropType<TreeNode>,
+      required: true,
+      default: () => ({}),
+    },
+  },
+
+  components: { VaCheckbox, VaIcon },
+
+  setup: (props, { slots }) => {
+    const {
+      nodeKey,
+      labelKey,
+      iconColor,
+      selectable,
+      colorComputed,
+      getKey,
+      toggleNode,
+      toggleCheckbox,
+    } = inject(TreeViewKey, {
```

Comment:
Maybe throw error if VaTreeNode used without VaTreeView? Or warn? Or we want user to be able to use VaTreeNode without VaTreeView?

Reply:
Recently I [implemented](https://github.com/epicmaxco/vuestic-ui/blob/develop/packages/ui/src/composables/useStrictInject.ts) `useStrictInject` composable, I'll need to merge an actual develop branch into the mine.